proposals/readonly-methods.md

```diff
@@ -0,0 +1,65 @@
+# Readonly Methods
```

Comment:
This is basically identical to the issue: https://github.com/dotnet/csharplang/issues/1710, with a couple minor changes I've called out below.